File: [23.12]Design-Pattern/Java/src/structural/bridge/Main.java
package structural.bridge;

public class Main {
    public static void main(String[] args) {
        Button startButton = new StartButton(new Red());
        Button endButton = new EndButton(new Blue());

        System.out.println("StartButton click");
        startButton.color.getColor();
        startButton.action();

        System.out.println("------------------------");

        System.out.println("EndButton click");
        endButton.color.getColor();
        endButton.action();
    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/structural/bridge/Main.kt
package structural.bridge

fun main() {
    val startButton = StartButton(Red())
    val endButton = EndButton(Blue())

    println("StartButton")
    startButton.color.getColor()
    startButton.action()

    println("----------------")

    println("StartButton")
    endButton.color.getColor()
    endButton.action()

}